ducktape/cluster/finite_subcluster.py

```diff
@@ -12,17 +12,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from ducktape.cluster.cluster import Cluster
-from ducktape.cluster.remoteaccount import RemoteAccount
+from ducktape.cluster.mixed_os_cluster import MixedOsCluster
 
 
-class FiniteSubcluster(Cluster):
+class FiniteSubcluster(MixedOsCluster):
     """This cluster class gives us a mechanism for allocating finite blocks of nodes from another cluster.
     """
     def __init__(self, nodes):
         self.nodes = nodes
-        self._available = set(self.nodes)
-        self._in_use = set()
+        self.available_nodes = set(self.nodes)
```

Comment:
Are these meant to be public? I don't see them used outside this class. In Python, you prefix with an `_` to indicate a field is intended to be private to the class. We haven't followed this strictly but have gradually been trying to migrate to this approach over time.